src/Visitor/NameResolver.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\SemanticVersionChecker\Visitor;
+
+use PhpParser\Node;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\NodeVisitor\NameResolver as ParserNameResolver;
+use PhpParser\BuilderHelpers;
+use PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocNode;
+use PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocTagNode;
+use PHPStan\PhpDocParser\Ast\Type\TypeNode;
+use PHPStan\PhpDocParser\Ast\Type\UnionTypeNode;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\ConstExprParser;
+use PHPStan\PhpDocParser\Parser\PhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use PHPStan\PhpDocParser\Parser\TypeParser;
+
+/**
+ * Extended Name Resolver that parse and resolve also docblock hintings
+ */
+class NameResolver extends ParserNameResolver
+{
+    /**
+     * @inheritDoc
+     */
+    public function enterNode(Node $node)
+    {
+        $return = parent::enterNode($node);
+
+        if ($node instanceof ClassMethod) {
```

Comment:
`Property`, `ClassConst` also may have annotation that describes a type. For Magento this is a less important cases and may be addressed in separate PRs.